model-optimizer/mo/main.py

```diff
@@ -313,15 +313,17 @@ def raise_ie_not_found():
     return argv
 
 
-def prepare_ir(argv):
+def prepare_ir(argv : argparse.Namespace):
     argv = arguments_post_parsing(argv)
 
     t = tm.Telemetry()
     graph = None
     ngraph_function = None
     moc_front_end, available_moc_front_ends = get_moc_frontends(argv)
-
-    if moc_front_end:
+    apply_fallback = False
```

Comment:
Can you wrap it in something like this?

```
def use_fallback(argv):
    def extensions():
        return hasattr(argv, 'extensions') and argv.extensions is not None and len(argv.extensions) > 0
    def transformations():
        return hasattr(argv, 'transformations_config') and argv.transformations_config is not None and len(argv.transformations_config) > 0

    return extensions() or transformations()

apply_fallback = use_fallback (argv)
```

You can even call it directly in the if statement in that case.

Reply:
@tomdol I've used the map here in order to use a key as info for events/logs